utils/src/array/mod.rs

```diff
@@ -25,14 +25,14 @@ pub enum Error<E: StdError + Send + Sync + 'static> {
     Other(E),
 }
 
-/// Types that can be fallibly read from a fixed-size byte sequence.
+/// Types that can be read from a variable-size byte sequence.
 ///
-/// `Array` is typically used to parse things like `PublicKeys` and `Signatures`
-/// from an untrusted network connection. Once parsed, these types are assumed
-/// to be well-formed (which prevents duplicate validation).
+/// `VarArray` is typically used to parse things like `Requests` from an untrusted
+/// network connection. Once parsed, these types are assumed to be well-formed
+/// (which prevents duplicate validation).
 ///
-/// If a byte sequencer is not properly formatted, `TryFrom` must return an error.
-pub trait Array:
+/// If a byte sequence is not properly formatted, `TryFrom` must return an error.
+pub trait VarArray:
```

Comment:
Not in love with the name but love the functionality 🤔

Reply:
Yeah in my head, `Array` is usually the variable-length one already...

I sort of prefer `Array` (variable) and `Fixed` (fixed-size)

Reply:
I wanted to keep `Array` as fixed-size because that aligns with Rust's definition.

Reply:
https://doc.rust-lang.org/std/primitive.array.html

```
A fixed-size array, denoted [T; N], for the element type, T, and the non-negative compile-time constant size, N.
```

Reply:
The closest rust-y thing would be `A contiguous growable array type, written as Vec<T>, short for ‘vector’` but people use `Vec` unlike `Array`, so that seems confusing as hell.

Reply:
We also don't allow it to grow (at least not in the trait def).